### PR TITLE
Update to FV description and associated pages in the Features area.

### PR DESCRIPTION
**Pull request by stevethesleeve**
Hi Nicolás,

Please would you look at this PR when you have a moment? Much of it reflects your prototype and we have probably reached a stage where it would be helpful to include these files in the main site, ready for further reviews as we proceed.

The FV function description has been revamped so that it hopefully reflects agreements during our discussions. I have also updated the fv.xlsx file to include some error examples, so it too is “finished for now” I hope.

I have included the updated Error Types page and the new Value Types, Optional Arguments and Units pages into the Features folders. I’ve also updated the config.mts file to get links on the left of the screen for these items. I’ve edited the material in the Optional Arguments page significantly but left the other three mostly as they are for now.

I have some other matters to discuss that arose during this update, but these are not related directly to this PR and so I’ll cover them in a separate email.

Regards,

Steve

### CODE REVIEW

File: docs/src/features/error-types.md [hunk 1]
```diff
@@ -7,73 +7,151 @@ lang: en-US
...
 
 ```
 =IF(B1=0, "N/A", A1/B1)
 ```
 
 ### **`#NAME?`**
 
-**Cause:** Unrecognized text in the formula (e.g., misspelled function names or undefined named ranges).  
-**Fix:** Correct spelling or define the missing name.
+Found when a name is not recognized. Maybe a misspeled name for a function. Could be a referenceto defined name that has been deleted.
```
Comment (nhatcher) [inline docs/src/features/error-types.md]: referenceto => reference to

File: docs/src/features/error-types.md [hunk 2]
```diff
@@ -7,73 +7,151 @@ lang: en-US
...
+
+Sometimes a `#NUM!` might be expected signalling the user that some parameter is out of scope.
 
 ### **`#N/A`**
 
-**Cause:** A value is not available, often in lookup functions like VLOOKUP.  
-**Fix:** Ensure the lookup value exists or use IFNA() to handle missing values:
+A value is not available, often in lookup functions like VLOOKUP.
+
+This is frequnly not an error in your spreadsheetlogic.
```
Comment (nhatcher) [inline docs/src/features/error-types.md]: frequnly => frequently

File: docs/src/features/error-types.md [hunk 3]
```diff
@@ -7,73 +7,151 @@ lang: en-US
...
 ### **`#NULL!`**
 
-**Cause:** Incorrect range operator in a formula (e.g., missing a colon between cell references).  
-**Fix:** Use correct range operators (e.g., A1:A10).
+Incorrect range operator in a formula (e.g., missing a colon between cell references).
 
+### **`#SPILL!`**
+
+A cell in a formula will overwrite content in other cells.
+This cannot happen riht now in IronCalc as formulas don't spill yet.
```
Comment (nhatcher) [inline docs/src/features/error-types.md]: riht => right

File: docs/src/features/error-types.md [hunk 4]
```diff
@@ -7,73 +7,151 @@ lang: en-US
...
 
 ### **`#CIRC!`**
 
-**Cause:** Circular reference.  
-**Fix:** Remove the circular reference.
+Circular reference. This is an error is your spreadsheet and must be fixed.
+Means that during teh course of a computation a circular dependency was found.
```
Comment (nhatcher) [inline docs/src/features/error-types.md]: teh => the

File: docs/src/features/error-types.md [hunk 5]
```diff
@@ -7,73 +7,151 @@ lang: en-US
...
+
+```
+=A1+
+```
+
+### **`#N/IMPL!`**
+
+A particular feature is not yet implemented in IronCalc
+
+Look if there is a Github ticket or contact us via email, Discord or bluesky
```
Comment (nhatcher) [inline docs/src/features/error-types.md]: I think I would leave out bluesky, email or Discord is fine. But I would add a link to both.

File: docs/src/functions/markdown-snippets/error-type-details.txt [hunk 6]
```diff
@@ -0,0 +1 @@
+* For more information about the different types of errors that you may encounter when using IronCalc functions, visit our [Error Types](/features/error-types) page.
```
Comment (nhatcher) [inline docs/src/functions/markdown-snippets/error-type-details.txt:1]: I don't think this works. On my branch I was trying not to have a markdown file because they also get translated to html, but the problem is that including txt files does not work at all.

**Review by nhatcher — CHANGES_REQUESTED** [review]
Hi @stevethesleeve,

Thanks for this. I found some typos, most likely done by me initially. Feel free to delete entire contents. Most of the things I wrote in my other branch are more of a _placeholder_ than actual content.

**Comment by stevethesleeve** [comment]
Hi @nhatcher, 

I'll sort out the typos tomorrow. Sorry I let them slip though - I was too busy getting the files into the system and inserting links to them inside fv.md!

Using the .txt for the snippet file seemed to work okay in my development server. I'm happy either way and so I'll revert it to the original .md extension for now.

:thumbsup:

**Comment by nhatcher** [comment]
> Using the .txt for the snippet file seemed to work okay in my development server. 

Really? Let me try again. If so it is not a bad solution

**Comment by stevethesleeve** [comment]
Hi @nhatcher ,

I've updating the Error Types and Value Types pages to fix the typos you mentioned. I've also read both pages again and tidied up a few things that we didn't spot before.

I know these files will need on-going maintenance but please let me know if you find any other issues that need urgent attention.

File: docs/src/functions/markdown-snippets/error-type-details.txt (hunk 6 above)
Comment (nhatcher) [reply docs/src/functions/markdown-snippets/error-type-details.txt:1]: My bad, just tested and this works perfectly well.

**Review by nhatcher — APPROVED** [review]
LGTM!

Thanks @stevethesleeve , I will deploy this later today
